.github/workflows/cd.yaml

```diff
@@ -0,0 +1,83 @@
+name: HOPR CD
+
+on:
+  push:
+    branches: ["debug/**"]
+    paths-ignore:
+      - "docs/**/*"
+      - "packages/*/docs/**/*"
+
+jobs:
+  cd:
+    name: Build HOPRD
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git info
+        run: ./scripts/configure-git-info.sh
+
+      - name: Setup Google Cloud Credentials
+        uses: google-github-actions/setup-gcloud@v0.2.1
+        with:
+          project_id: ${{ secrets.GOOGLE_HOPRASSOCIATION_PROJECT }}
+          service_account_key: ${{ secrets.GOOGLE_HOPRASSOCIATION_CREDENTIALS_REGISTRY }}
+          export_default_credentials: true
+
+      - name: Set image version
+        id: set-image-version
+        run: echo "::set-output name=vsn::$(date +%s)"
+
+      - name: Set package version
+        id: set-package-version
+        run: echo "::set-output name=vsn::$(./scripts/get-package-version.sh)"
+        env:
+          HOPR_PACKAGE: hoprd
+
+      - name: Building Docker image using Google Cloud Build
+        run: |
+          gcloud builds submit --config cloudbuild.yaml \
+            --substitutions=_HOPR_PACKAGE_VERSION=${HOPR_PACKAGE_VERSION},_HOPR_IMAGE_VERSION=${HOPR_IMAGE_VERSION}
+        env:
+          HOPR_PACKAGE: hoprd
+          HOPR_DOCKER_IMAGE: gcr.io/hoprassociation/hoprd
+          HOPR_IMAGE_VERSION: ${{ steps.set-image-version.outputs.vsn }}
+          HOPR_PACKAGE_VERSION: ${{ steps.set-package-version.outputs.vsn }}
+
+      - name: Verify the correct hoprd package version has been bundled
+        run: |
+          declare v=$(docker run --pull always ${HOPR_DOCKER_IMAGE}:${HOPR_IMAGE_VERSION} --version 2> /dev/null)
+          [ "${v}" = "${HOPR_PACKAGE_VERSION}" ]
+        env:
+          HOPR_DOCKER_IMAGE: gcr.io/hoprassociation/hoprd
+          HOPR_IMAGE_VERSION: ${{ steps.set-image-version.outputs.vsn }}
+          HOPR_PACKAGE_VERSION: ${{ steps.set-package-version.outputs.vsn }}
+
+      - name: Run start script
+        if: ${{ !env.ACT }}
+        run: |
+          echo "Pre-deploy env var RELEASE=${RELEASE}"
+          ./scripts/deploy-machine.sh
+        env:
+          FUNDING_PRIV_KEY: ${{ secrets.FUNDING_WALLET_PRIVATE_KEY }}
+          FAUCET_SECRET_API_KEY: ${{ secrets.FAUCET_SECRET_API_KEY }}
+          INFURA_KEY: ${{ secrets.INFURA_KEY }}
+          GITHUB_REF: ${{ github.ref }}
+          BS_PASSWORD: ${{ secrets.BS_PASSWORD }}
+          RELEASE: ${{ steps.set-image-version.outputs.vsn }}
+          HOPRD_API_TOKEN: ${{ secrets.HOPRD_API_TOKEN }}
```

Comment:
Could you make the secret name more specific to its use. E.g. `secrets.HOPRD_PR_PREVIEW_API_TOKEN` or so? That way we can have different token secrets for the various clusters.

Reply:
I believe we don't even use that variable, but have it hardcoded in `deploy-machine.sh`. Also, I'm not sure whether we would like different token secrets for our pr-preview vms, especially since these ones will be dead after 24hrs. I figure this would make things harder to test in the long run.